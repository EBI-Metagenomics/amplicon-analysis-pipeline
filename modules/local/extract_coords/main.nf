
process EXTRACT_COORDS {
    tag "$meta.id"
    label 'light'
    conda "${projectDir}/conf/environment.yml"

    input:
    tuple val(meta), path(easel_coords)
    tuple val(meta), path(matched_seqs_with_coords)

    output:
    tuple val(meta),  path("sequence-categorisation/*SSU.fasta"), optional: true, emit: ssu_fasta
    tuple val(meta),  path("sequence-categorisation/*LSU.fasta"), optional: true, emit: lsu_fasta
    tuple val(meta),  path("*concat_SSU_LSU_coords"), optional: true, emit: concat_ssu_lsu_coords


    """
    get_subunits -i $easel_coords -n ${meta.id}
    get_subunits_coords -i $matched_seqs_with_coords -s SSU -l LSU
    cat SSU_coords LSU_coords > ${meta.id}_concat_SSU_LSU_coords
    """

}
