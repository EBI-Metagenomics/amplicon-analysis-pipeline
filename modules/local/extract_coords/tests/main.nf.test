nextflow_process {

    name "Test Process EXTRACT_COORDS"
    script "../main.nf"
    process "EXTRACT_COORDS"
    tag "modules"
    tag "extract_coords"

    test("Should take a fasta file and an easel coordinates file and extract two subunit fasta files for the SSU and LSU, and a concatenated SSU and LSU coordinates file") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('${projectDir}/modules/local/extract_coords/tests/data/test.fasta', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file('${projectDir}/modules/local/extract_coords/tests/data/test.matched_seqs_with_coords', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.ssu_fasta).match("ssu_fasta") },
                { assert snapshot(process.out.lsu_fasta).match("lsu_fasta") },
                { assert snapshot(process.out.concat_ssu_lsu_coords).match("concat_ssu_lsu_coords") }
            )
        }

    }

}
