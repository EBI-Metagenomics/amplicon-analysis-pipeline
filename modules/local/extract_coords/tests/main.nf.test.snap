{
    "concat_ssu_lsu_coords": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_concat_SSU_LSU_coords:md5,401222d263116bb145c1a67e6acc4709"
                ]
            ]
        ],
        "timestamp": "2023-11-02T13:55:30.261913771"
    },
    "lsu_fasta": {
        "content": [
            [
                
            ]
        ],
        "timestamp": "2023-11-02T13:55:30.248331773"
    },
    "ssu_fasta": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_SSU.fasta:md5,fb23f13526672a300845b767dbc5fd44"
                ]
            ]
        ],
        "timestamp": "2023-11-02T13:55:30.213241815"
    }
}